### ISSUE TITLE
Proxy for package manager

### ISSUE THREAD

**Issue by FabiJa** [OpenModelica/OpenModelica#10648]
Hi,

I am using OM under the WSL, which works greats so far. But I have a problem using the package manager. I am behind a proxy, but the corresponding env variables are set. If I want to install a new library I get an error `Curl error for URL https://github.com/*.zip Couldn't resolve host name`. If I try to download the zip file in the WSL terminal, it works.
In addition, the OM tab "Latest News & Events" says "Sorry, no internet no news items.". I am quite sure, that I had no connection issue with some older OM version, but cannot reproduce this.

Any advice?

Thx
Fabian

**Comment by casella** [comment]
@sjoelund any comment on this?

**Comment by sjoelund** [comment]
What environment variables are used? Curl can either lookup hostnames using the proxy or the host DNS:

https://unix.stackexchange.com/questions/175888/curl-7-27-any-proxy-set-curl-does-not-resolve-the-hostname-via-proxy

**Comment by FabiJa** [comment]
Env variables are `http_proxy` and `https_proxy`.

**Comment by sjoelund** [comment]
Try the `ALL_PROXY` environment variable. https://curl.se/libcurl/c/CURLOPT_PROXYTYPE.html needs CURLPROXY_SOCKS5_HOSTNAME set and I guess http_proxy and https_proxy does not set it

**Comment by casella** [comment]
Side note: if this eventually works, we should report it in the [download-linux](https://openmodelica.org/download/download-linux/) page.

**Comment by FabiJa** [comment]
I set `ALL_PROXY=http://proxy_ip:port`. This did not help, although I am not sure if this is the modification you ment.
I am also wondering why this should be necessary, when `curl` is working from the shell.

**Comment by FabiJa** [comment]
Update:
I figured out that the problem results from starting OMEdit (WSL) via the Windows Start menu. If I start it from the WSL terminal with `OMEdit` the package manager is working and I can install packages. The symptomes are loosly related to this [issue](https://github.com/microsoft/wslg/issues/1052). Maybe the call defined in the OMEdit (WSL) shortcut is not correct or an issue with WSL itself.

**Comment by sjoelund** [comment]
Yes, running wsl.exe seems to ignore most environment variables... https://stackoverflow.com/questions/52504895/wsl-cant-see-environment-variable

**Comment by casella** [comment]
@FabiJa it seems you have some experience at running OMEdit under WSL. Would you be interested at writing down a guide on how to do that, where you report what you learned?

**Comment by FabiJa** [comment]
> Yes, running wsl.exe seems to ignore most environment variables (and bashrc/profile)... https://stackoverflow.com/questions/52504895/wsl-cant-see-environment-variable

Since we had the same problem with Windows, there it helped to set `ALL_PROXY=http://proxy_ip:port`.

**Comment by FabiJa** [comment]
> @FabiJa it seems you have some experience at running OMEdit under WSL. Would you be interested at writing down a guide on how to do that, where you report what you learned?

Is there a template for this, so I can estimate the effort?

**Comment by casella** [comment]
> Is there a template for this, so I can estimate the effort?

Not really. I would put the instructions in the [Download Windows page](https://openmodelica.org/download/download-windows/). Keep them short and simple. The sources of the webpage are found in this repo: https://github.com/OpenModelica/www.openmodelica.org, you can open a PR that adds some text to https://github.com/OpenModelica/www.openmodelica.org/blob/main/content/download/download-windows.md

**Comment by FabiJa** [comment]
> > Is there a template for this, so I can estimate the effort?
> 
> Not really. I would put the instructions in the [Download Windows page](https://openmodelica.org/download/download-windows/). Keep them short and simple. The sources of the webpage are found in this repo: https://github.com/OpenModelica/www.openmodelica.org, you can open a PR that adds some text to https://github.com/OpenModelica/www.openmodelica.org/blob/main/content/download/download-windows.md

Ok, I will check this out next week. Overall using the GUI (OMEdit) has still some glitches if using WSLg, especially if another windows is opened (like information on for a model class). You neither can close nor move it.

Regarding the original problem of ignored env variables I found this [post ](https://superuser.com/a/1714780), which is at least a work around, which works for me i.e. modifying the windows shortcut to `wslg.exe -d your_distro bash -ilc OMEdit`.
